components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -137,19 +209,20 @@ impl CanvasRenderingContext2D {
         script_to_constellation_chan
             .send(ScriptMsg::CreateCanvasPaintThread(size, sender))
             .unwrap();
-        let (ipc_renderer, canvas_id) = receiver.recv().unwrap();
+        //let (canvas_state.ipc_renderer, canvas_id) = receiver.recv().unwrap();
```

Comment:
This code which creates a new paint thread should be shared between both the offscreen and onscreen canvases.

Reply:
How do I do that?

Reply:
I believe it should be moved to the CanvasState constructor.

Reply:
> I believe it should be moved to the CanvasState constructor.

Isn't it already in CanvasState constructor?

Reply:
The code that creates a new canvas paint thread currently exists in the CanvasRenderingContext2D code, right above this code that we're discussing.